src/config.rs

```diff
@@ -28,87 +42,154 @@ impl Config {
-        let reposdir = match canonicalize(&reposdir_str) {
-            Ok(p) => match p.to_str() {
-                Some(s) => s.to_owned(),
-                None => error(&format!("{}: can't convert to string", &reposdir_str)),
-            },
-            Err(e) => {
-                error(&format!("{}: {}", &reposdir_str, e));
+                            let reposdir_str = lexer.lexeme_str(&lexeme);
+                            let reposdir_str = &reposdir_str[1..reposdir_str.len() - 1];
+                            reposdir = Some(match canonicalize(reposdir_str) {
+                                Ok(p) => match p.to_str() {
+                                    Some(s) => s.to_owned(),
+                                    None => fatal(&format!(
+                                        "'{}': can't convert to string",
+                                        &reposdir_str
+                                    )),
+                                },
+                                Err(e) => {
+                                    fatal_err(&format!("'{}'", reposdir_str), e);
+                                }
+                            });
+                        }
+                        GenericOption::Secret(lexeme) => {
+                            if secret.is_some() {
+                                conf_fatal(
+                                    &lexer,
+                                    lexeme,
+                                    "Mustn't specify 'secret' more than once",
+                                );
+                            }
+                            let secret_str = lexer.lexeme_str(&lexeme);
+                            let secret_str = &secret_str[1..secret_str.len() - 1];
+                            secret = Some(SecStr::from(secret_str));
+                        }
+                    }
+                }
             }
-        };
-
-        let secret = {
-            let p = &matches
-                .opt_str("s")
-                .unwrap_or_else(|| error("A secrets path must be specified."));
-            let path = Path::new(p);
-            SecStr::new(
-                read_to_string(path)
-                    .unwrap_or_else(|_| error("Couldn't read secrets file."))
-                    .trim()
-                    .to_owned()
-                    .into_bytes(),
-            )
-        };
+            _ => process::exit(1),
+        }
+        if maxjobs.is_none() {
+            maxjobs = Some(num_cpus::get());
+        }
+        if port.is_none() {
+            fatal("A port must be specified");
+        }
+        if reposdir.is_none() {
+            fatal("A directory for per-repo programs must be specified");
+        }
+        if secret.is_none() {
+            fatal("A secret must be specified");
+        }
 
         Config {
-            maxjobs,
-            port,
-            reposdir,
-            secret,
+            maxjobs: maxjobs.unwrap(),
```

Comment:
If you unwrap your config directives above, you can drop this hunk. E.g.:

```
let port = port.unwrap_or_else(|| fatal(...));
```

Reply:
True. I'm not sure it's an improvement or not. Maybe I might change this in a future PR when I've slept on it!

Reply:
It's minor really. I don't care too much.